frontend/modules/presentation/presentation.sagas.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ *  Copyright (C) 2020 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { put, select, takeLatest} from 'redux-saga/effects';
+
+import { CHAT_CHANNELS } from '../../constants/chat';
+
+import * as API from '../../services/api';
+import { Viewer } from '../../services/viewer/viewer';
+import { ChatActions } from '../chat';
+import { selectCurrentModel, selectCurrentModelTeamspace } from '../model';
+import { dispatch, getState } from '../store';
+import {  selectIsPaused, selectSessionCode, PresentationActions, PresentationTypes } from './index';
+
+let intervalId = 0;
+
+const streamPresentation = async (teamspace, model, code) => {
+	const viewpoint = await Viewer.getCurrentViewpoint({ teamspace, model });
```

Comment:
(Once it's ready) it should be using the same code as issues/risks/views to get the viewpoint. The idea is that we should get the highlighting/hidden objects/colour override and any future new features in viewpoint for free